drivers/SmartThings/matter-appliance/src/matter-refrigerator/init.lua

```diff
@@ -239,21 +268,18 @@ local function handle_refrigerator_tcc_mode(driver, device, cmd)
 end
 
 local function handle_temperature_setpoint(driver, device, cmd)
-  local tn_eps = embedded_cluster_utils.get_endpoints(device,clusters.TemperatureControl.ID, {feature_bitmap = clusters.TemperatureControl.types.Feature.TEMPERATURE_NUMBER})
-  if #tn_eps == 0 then
-    device.log.warn_with({ hub_logs = true }, string.format("Device does not support TEMPERATURE_NUMBER feature"))
+  local ep = component_to_endpoint(device, cmd.component)
+  if not supports_temperature_number_endpoint(device, ep) then
```

Comment:
You can probably remove this also. There shouldn't be a situation where the app is sending a temperature setpoint command for an endpoint that doesn't support the temperature number. The `component_to_endpoint` function will make sure that the component on the app is mapped to the correct endpoint on the device so that if the component on the app (the button the user presses) supports temperatureNumber, then the endpoint that maps to that component should also support temperatureNumber if we have done the endpoint to component mapping correctly

Reply:
We can't know if a manufacturer actually supports Matter correctly. I think we should be prepared even when they are not applying properly. We confirmed such case with Virtual device app.